specification/src/chapter3.adoc

```diff
@@ -450,6 +450,80 @@ and vector cryptography extensions.
 
 The High Assurance Cryptography task group will create instruction set extensions (ISEs) that facilitate higher levels of assurance than the existing Scalar and Vector Crypto ISEs. One initial focus will be on full-rounds vector AES extensions that allow (do not prevent) effective side-channel resistant implementations and that may perform better than the existing round-based instructions, with future work on other algorithms. A second intimately related focus area will be ISEs that manage secret keys -- not restricted to just AES keys -- in ways that better protect them from unauthorized users and from side-channel analysis.
 
+=== Architectural metadata storage
+
+Some RISC-V architectural features require storage for metadata - _architectural metadata storage_. For example:  MTT metadata, metadata for memory tagging, or CHERI metadata.
+
+Architectural metadata storage is implementation defined, but the following rules should be considered by any implementation. 
+
+See xref:chapter2.adoc#_adversarial_model[adversarial model] 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| SR_AMS_001
+| Architectural metadata storage MUST be protected against logical attacks
+
+| SR_AMS_002
+| Architectural metadata storage MUST be protected against physical attacks
+
+| SR_AMS_003
+| Architectural metadata storage MUST be protected against direct attacks
+
+|===
+
+For example, architectural metadata storage may be implemented in on-chip memory, or in cryptographically protected external DDR.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| SR_AMS_004
+| Architectural metadata MUST be isolated by privilege level, and along isolation boundaries
+
+|===
+
+Depending on use case, architectural metadata may be visible to or managed by, for example, a supervisor level kernel, a kernel or a hypervisor within a supervisor domain, or by a machine mode monitor. But it should be considered private within an isolation boundary and not accessible or guessable by lower privilege levels or by code in a different supervisor domain.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| SR_AMS_005
+| Cryptographically protected architectural metadata storage MUST provide privacy protection
+
+| SR_AMS_006
+| Cryptographically protected architectural metadata storage MUST provide integrity protection
+
+| SR_AMS_007
+| Cryptographically protected architectural metadata storage SHOULD provide replay protection
+
+|===
+
+Architectural metadata needs to be protected against both unauthorized access, and errors (accidental or malicious).
+
+On systems where architectural metadata is stored in external memory, and external memory attacks are in scope (for example, directly accessible or replacable external memory), then cryptographic replay protection is strongly recommended.
```

Comment:
Typo in `replacable`

Reply:
'access' here includes reading and modifying ? 

also need to consider data relocation attacks when stored in external memory.  encryption must be location dependent, freshness protected, and authenticated.

Reply:
Fixed "replaceable".

Added location freshness to SR_AMS_005.

Added temporal freshness to SR_AMS_006.

Clarified the rationale paragraphs.